onnxruntime/core/framework/data_transfer_manager.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/status.h"
+#include "core/framework/data_transfer.h"
+#include "core/framework/tensor.h"
+
+namespace onnxruntime {
+
+// Data transfer manager, which has all functions registered to copy tensors with different location.
+// It's not thread-safe.
+class DataTransferManager {
```

Comment:
Looks like this class is serving 2 purposes - act as the registry of data transfer instances and as a facade to facilitate the transfers (via the CopyTensor method). Wondering if it makes sense to separate these responsibilities.

Reply:
that's correct. I think it's ok to have the two in the same class now given the complexity.

Reply:
DataTransferUtil might be a better class name for separation to act as the facade. It should use the DataTransferManager singleton.

Reply:
I agree with @pranavsharma and would prefer splitting the registration duties from the copying. First rule of 'SOLID' is that a class should have a single responsibility. https://en.wikipedia.org/wiki/SOLID.

A slightly more standard setup would be to retrieve the registered object that does the work, and it would perform the copy duties.